third_party/terraform/resources/resource_storage_bucket.go

```diff
@@ -632,10 +644,26 @@ func resourceStorageBucketRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("website", flattenBucketWebsite(res.Website))
 	d.Set("retention_policy", flattenBucketRetentionPolicy(res.RetentionPolicy))
 
-	if res.IamConfiguration != nil && res.IamConfiguration.BucketPolicyOnly != nil {
-		d.Set("bucket_policy_only", res.IamConfiguration.BucketPolicyOnly.Enabled)
-	} else {
-		d.Set("bucket_policy_only", false)
+	// Use this in the next major version of the provider.
+	// if res.IamConfiguration != nil && res.IamConfiguration.UniformBucketLevelAccess != nil {
+	// 	d.Set("uniform_bucket_level_access", res.IamConfiguration.UniformBucketLevelAccess.Enabled)
+	// } else {
+	// 	d.Set("uniform_bucket_level_access", false)
+	// }
+
+	// Delete this block in next major version of the provider.
+	if _, ok := d.GetOk("bucket_policy_only"); ok {
```

Comment:
the "ok" in GetOk returns whether or not it's set to a non-zero value, rather than if it's been set at all. So these blocks will only work if bucket_policy_only/uniform_bucket_level_access are set to true in the user's config. Does Terraform complain if we unconditionally set both fields?

Reply:
Also, if the API accepts either field and returns the same value in both, then we can simplify the code here and in the expander by just using `UniformBucketLevelAccess`.